tests/integration/test_ethernet.py

```diff
@@ -61,6 +61,15 @@ def provider(request):
 def testnic1():
     veth_name = "testeth"
     try:
+        if os.path.isfile("/etc/sysconfig/network-scripts/ifcfg-" + veth_name):
+            os.unlink("/etc/sysconfig/network-scripts/ifcfg-" + veth_name)
+            subprocess.call(
+                [
+                    "nmcli",
+                    "con",
+                    "reload",
+                ]
+            )
```

Comment:
does `nmcli con reload` return immediately, or does it wait for the connection(s) to be reloaded?

Reply:
AFAIK, it returns immediately.

Reply:
Then is there a chance that the `ip link add testeth` call might fail if `nmcli con reload` has not completed?

Reply:
If at all, the cleanup needs to happen in the `finally` block.

Reply:
> Then is there a chance that the `ip link add testeth` call might fail if `nmcli con reload` has not completed?

I honestly did not see the racing problem between the `ip link add testeth` and `nmcli con reload`.